luigi/contrib/sqla.py

```diff
@@ -328,15 +334,37 @@ def construct_sqla_columns(columns):
         else:
             # if columns is specified as (name, type) tuples
             with engine.begin() as con:
-                metadata = sqlalchemy.MetaData()
+
+                if self.schema:
+                    curs = engine.execute(sqlalchemy.text("""
+                        SELECT DISTINCT schema_name
+                        FROM information_schema.schemata
+                        WHERE schema_name = '{}'
+                    """.format(self.schema)))
+                    # If this is not None then the schema already exists
+                    row = curs.fetchone()
+                    if row is not None:
+                        self.logger.info(
+                            'Schema {} already exists, not creating'.format(
+                                self.schema,
+                            )
+                        )
+                    if not row:
+                        self.logger.info('Creating schema {}'.format(self.schema))
+                        engine.execute(sqlalchemy.schema.CreateSchema(self.schema))
```

Comment:
Yes, it seems quite reasonable to ask users to create the schema separately to avoid this kind of silent failure. I'll remove that part and add a note.